src/PAModelTests/EntropyTests.cs

```diff
@@ -154,5 +155,101 @@ public void TestPCFControlWillFallBackToControlTemplate(string appName)
                 Assert.IsTrue(MsAppTest.Compare(root, tempFile.FullPath, Console.Out));
             }
         }
+
+        // Validate that the host control template hostType value is stored in entropy while unpacking
+        [DataTestMethod]
+        [DataRow("HostControlTestWithHostType.msapp")]
+        public void TestHostControlWithHostType(string appName)
+        {
+            var root = Path.Combine(Environment.CurrentDirectory, "Apps", appName);
+            Assert.IsTrue(File.Exists(root));
+
+            (var msapp, var errors) = CanvasDocument.LoadFromMsapp(root);
+            errors.ThrowOnErrors();
+
+            Assert.IsTrue(msapp._entropy.HostTypeEntry.Count > 0);
+
+            Assert.IsTrue(msapp._screens.TryGetValue("App", out var app));
+            var sourceFile = IRStateHelpers.CombineIRAndState(app, errors, msapp._editorStateStore, msapp._templateStore, new UniqueIdRestorer(msapp._entropy), msapp._entropy);
+
+            Assert.AreEqual("Host", sourceFile.Value.TopParent.Children.Last().Name);
+            // Checking if the HostType Entry is added
+            Assert.IsTrue(sourceFile.Value.TopParent.Children.Last().Template.ExtensionData.ContainsKey(IRStateHelpers.ControlTemplateHostTypePropertyName));
+            Assert.IsFalse(sourceFile.Value.TopParent.Children.Last().Template.ExtensionData.ContainsKey(IRStateHelpers.ControlTemplateHostServicePropertyName));
+        }
+
+        [DataTestMethod]
+        [DataRow("HostControlTestWithHostTypeAndHostService.msapp")]
+        public void TestHostControlWithHostTypeAndService(string appName)
+        {
+            var root = Path.Combine(Environment.CurrentDirectory, "Apps", appName);
+            Assert.IsTrue(File.Exists(root));
+
+            (var msapp, var errors) = CanvasDocument.LoadFromMsapp(root);
+            errors.ThrowOnErrors();
+
+            Assert.IsTrue(msapp._entropy.HostTypeEntry.Count > 0);
+            Assert.IsTrue(msapp._entropy.HostServiceEntry.Count > 0);
+
+            Assert.IsTrue(msapp._screens.TryGetValue("App", out var app));
+            var sourceFile = IRStateHelpers.CombineIRAndState(app, errors, msapp._editorStateStore, msapp._templateStore, new UniqueIdRestorer(msapp._entropy), msapp._entropy);
+
+            Assert.AreEqual("Host", sourceFile.Value.TopParent.Children.Last().Name);
+            // Checking if the HostType and HostService Entries were added
+            Assert.IsTrue(sourceFile.Value.TopParent.Children.Last().Template.ExtensionData.ContainsKey(IRStateHelpers.ControlTemplateHostTypePropertyName));
+            Assert.IsTrue(sourceFile.Value.TopParent.Children.Last().Template.ExtensionData.ContainsKey(IRStateHelpers.ControlTemplateHostServicePropertyName));
+        }
+
+        [DataTestMethod]
+        [DataRow("HostControlTestWithHostType.msapp")]
+        public void TestHostControlWithHostTypeFallBackToTemplate(string appName)
```

Comment:
Do these tests fail before your fix? Just looking at this, I'm not sure I understand how these tests would be passing since the entropy data was the only place this info was stored, and it has been cleared out.

Reply:
Good catch! Okay, so looks like this app needs some work. Let me look into it.